turbine/src/cluster_nodes.rs

```diff
@@ -273,8 +274,15 @@ impl ClusterNodes<RetransmitStage> {
         // Unstaked nodes' position in the turbine tree is not deterministic
         // and depends on gossip propagation of contact-infos. Therefore, if
         // this node is not staked return None.
-        if self.nodes[self.index[&self.pubkey]].stake == 0 {
-            return Ok(None);
+        {
+            // dedup_tvu_addrs might exclude a non-staked node from self.nodes
+            // due to duplicate socket/IP addresses.
+            let Some(&index) = self.index.get(&self.pubkey) else {
+                return Ok(None);
+            };
+            if self.nodes[index].stake == 0 {
+                return Ok(None);
+            }
```

Comment:
There is another iterator looking for own pubkey on line 296 (former 288)

Reply:
I think that is already covered.

Reply:
Ok assuming index stays in sync with nodes yes, should be fine.